src/saveload/marker.rs

```diff
@@ -0,0 +1,213 @@
+//! Provides `Marker` and `MarkerAllocator` traits
+
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::hash::Hash;
+
+use shred::Resource;
+use {Component, DenseVecStorage, Entities, Entity, Join, ReadStorage, WriteStorage};
+
+use serde::ser::Serialize;
+use serde::de::DeserializeOwned;
+
+
+/// This trait should be implemetened by component which is gonna be used as marker.
+/// This marker should be set to entity that should be serialized.
+/// If serialization strategy needs to set marker to some entity it should use
+/// new marker allocated for `Marker::Allocator`.
```

Comment:
s/it should use new marker allocated for/then it should use newly allocated marker from/